File: 01b.pl
#! /usr/bin/perl
use warnings;
use strict;
use feature qw{ say };

use ARGV::OrDATA;
use List::Util qw{ sum };

local $/ = "";

my @max = (0) x 3;
while (<>) {
    my $sum = sum(split /\s+/);
    if ($sum >= $max[0]) {
        push @max, $sum;
        @max = sort { $a <=> $b } @max;
        shift @max;
    }
}

say sum(@max);


__DATA__
1000
2000
3000

4000

5000
6000

7000
8000
9000

10000

File: 01b.py
#!/usr/bin/python3
import sys

maxes = [0, 0, 0]
with open(sys.argv[1], 'r') as fh:
    elf = 0
    for line in fh:
        if len(line) > 1:
            elf += int(line.rstrip('\n'))
        else:
            if elf > maxes[0]:
                maxes.append(elf)
                maxes.sort()
                maxes.pop(0)
            elf = 0
print(sum(maxes))
